From 565b5cd51abb8dca2690406aeab6d9da45812120 Mon Sep 17 00:00:00 2001
From: Jeff Moyer <jmoyer@redhat.com>
Date: Fri, 10 Oct 2014 21:03:10 -0400
Subject: [block] provide compat ioctl for BLKZEROOUT

Message-id: <1412975015-5370-10-git-send-email-jmoyer@redhat.com>
Patchwork-id: 97461
O-Subject: [RHEL7 PATCH 09/34] block: provide compat ioctl for BLKZEROOUT
Bugzilla: 1146660
RH-Acked-by: David Milburn <dmilburn@redhat.com>
RH-Acked-by: Mike Snitzer <snitzer@redhat.com>

This is a backport of the following commit.  This resolves bug
1146660.

  commit 3b3a1814d1703027f9867d0f5cbbfaf6c7482474
  Author: Mikulas Patocka <mpatocka@redhat.com>
  Date:   Wed Jul 2 12:46:23 2014 -0400

    block: provide compat ioctl for BLKZEROOUT

    This patch provides the compat BLKZEROOUT ioctl. The argument is a pointer
    to two uint64_t values, so there is no need to translate it.

    Signed-off-by: Mikulas Patocka <mpatocka@redhat.com>
    Cc: stable@vger.kernel.org # 3.7+
    Acked-by: Martin K. Petersen <martin.petersen@oracle.com>
    Signed-off-by: Jens Axboe <axboe@fb.com>

Signed-off-by: Jeff Moyer <jmoyer@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/block/compat_ioctl.c b/block/compat_ioctl.c
index 91f3706..e4fe4c8 100644
--- a/block/compat_ioctl.c
+++ b/block/compat_ioctl.c
@@ -690,6 +690,7 @@ long compat_blkdev_ioctl(struct file *file, unsigned cmd, unsigned long arg)
  case BLKROSET:
  case BLKDISCARD:
  case BLKSECDISCARD:
+ case BLKZEROOUT:
  /*
   * the ones below are implemented in blkdev_locked_ioctl,
   * but we call blkdev_ioctl, which gets the lock for us
-- 
1.7.1